src/frameworks/next/utils.ts

```diff
@@ -549,3 +550,59 @@ export function installEsbuild(version: string): void {
     }
   }
 }
+
+/**
+ * Create a PrerenderManifest["routes"] object from PagesManifest
+ */
+export function createPagesManifestLikePrerender(
+  pagesManifestJSON: PagesManifest,
+): PrerenderManifest["routes"] {
+  const pagesManifestLikePrerender: PrerenderManifest["routes"] = {};
+  for (const [path, route] of Object.entries(pagesManifestJSON)) {
+    if (typeof route === "string" && route.endsWith(".html")) {
+      pagesManifestLikePrerender[path] = {
+        srcRoute: null,
+        initialRevalidateSeconds: false,
+        dataRoute: "",
+        experimentalPPR: false,
+        prefetchDataRoute: "",
+      };
+    }
+  }
+  return pagesManifestLikePrerender;
+}
+
+/**
+ * Check if an HTML file is partial (doesn't have closing </html> tag)
+ */
+export async function isPartialHTML(filePath: string): Promise<boolean> {
```

Comment:
q, in my test project I see a *.prefetch.rsc file next to PPR content in the dist dir, is that a suitable test that wouldn't require reading the file?

Reply:
The problem with using the `*.prefetch.rsc` file is that it is present even for files that are non-partial html. If we use that as a test then we would be skipping valid (fully generated) html files.

Reply:
So digging in a bit on an experimental project, it seems as though the best test is the "postponed" key in the `.meta` file, we're already reading that for headers. Mind refactoring to use the presence of that key, rather than reading the HTML? That will also get us PPR for data routes, etc.

Reply:
Nice! Sorry for not noticing that property before. Just refactored it to use that property from the .meta files.